cluster-autoscaler/estimator/binpacking_estimator.go

```diff
@@ -89,14 +89,19 @@ func (e *BinpackingNodeEstimator) Estimate(
 	scheduledPods := []*apiv1.Pod{}
 	lastNodeName := ""
 
-	for _, pod := range pods {
-		found := false
+	for _, podsEquivalenceGroup := range podsEquivalenceGroups {
```

Comment:
Can you split this into smaller functions? It's pretty long right now, long nested loops are hard to read and there's a bit of code duplication that perhaps could be avoided.